### PR TITLE
Made the "Drafts and Posts folder" configurable via the options page.…

### PR DESCRIPTION
**Pull request by poppastring**
… It will take the current location of the "Drafts and Posts folder" location from the registry and make it part of the PostEditorSettings. After this it will always get the settings from PostEditor.

This should help resolve issue #29.

### CODE REVIEW

File: src/managed/OpenLiveWriter.PostEditor/PostEditorPreferencesPanel.cs [hunk 1]
```diff
@@ -114,6 +125,16 @@ public PostEditorPreferencesPanel()
 
         }
 
+        private void ButtonBrowserDialog_MouseClick(object sender, MouseEventArgs e)
+        {
+            DialogResult result = folderBrowserDialog.ShowDialog();
```
Comment (ScottIsAFool) [inline src/managed/OpenLiveWriter.PostEditor/PostEditorPreferencesPanel.cs]: When used, this is only allowing you to choose a folder that's inside your documents folder. There's no way of browsing your full PC.

**Comment by ScottIsAFool** [comment]
I'm also going to be really picky and request that the alignments of the new controls be adjusted. As you can see with the attached image, they're slightly off
writeralignment

**Comment by pekiZG** [comment]
Is this duplicate?

**Comment by pekiZG** [comment]
https://github.com/OpenLiveWriter/OpenLiveWriter/issues/29

**Comment by ScottIsAFool** [comment]
No, this is a pull request to fix #29

File: src/managed/OpenLiveWriter.PostEditor/PostEditorPreferencesPanel.cs (hunk 1 above)
Comment (poppastring) [inline src/managed/OpenLiveWriter.PostEditor/PostEditorPreferencesPanel.cs]: Thanks for the feedback will make that change.

**Comment by ScottIsAFool** [comment]
When I move the folder to a brand new folder and click apply, it's throwing an exception. It looks like we need to be creating the Drafts/Recent folders when we click apply. This should be done before `PostListCache.Update()` is called in PostEditorPreferencesPanel.cs.

I'm also wondering whether, when the default folder is changed, we should be moving posts from the old folder to the new folder? Perhaps a question to be asked for the user if they change this (same as Windows does if you change the location of downloads/documents/pictures/etc)

**Comment by poppastring** [comment]
Got the exception fixed, also there is now an option to move the existing posts to the new folder location.

**Comment by ScottIsAFool** [comment]
Awesome, will check it out

**Comment by ScottIsAFool** [comment]
Ok, so I ran into one small problem. Steps to repro:
1. Open Live Writer
2. Open options
3. Change location and apply (move contents if required)
4. Make some changes to the current fresh blog post
5. Save. 

This stores that draft in the old folder, not the new. If you make the folder change while you have an existing draft already open, this doesn't seem to happen.

Also, is it possible when you click on browse to open the folder picker on that folder already?

**Comment by ScottIsAFool** [comment]
Hey @poppastring, I have submitted a PR to your fork with the changes I suggested above. It also includes a merge of everything in Master right now so that it was all up to date.

**Comment by poppastring** [comment]
Thanks for the follow up, I have pulled in your changes as requested @ScottIsAFool

**Comment by ScottIsAFool** [comment]
Excellent, thanks. Now this just needs some testing (other than me :)) @kathweaver

**Comment by kathweaver** [comment]
LGT Though I need to make some changes to the test plans, and I haven't figured out the best way yet.

**Comment by kathweaver** [comment]
I do see one problem.  If I change where I save the existing posts, then go to open an existing post, the last post I saved is still showing in File | Open Draft when I hover.  If I try to open that post I get a file not found.

It will only happen if I hover.  If I click on any of those, it refreshes and works correctly by showing no files.

**Comment by ScottIsAFool** [comment]
Good shout, didn't think of changing that. That might also affect the jump list too.

**Comment by ScottIsAFool** [comment]
@poppastring will you get a chance to look at @kathweaver's points?

**Comment by poppastring** [comment]
@ScottIsAFool Yep, I will take a look tonight, I will let you know if I run into any problems.

**Comment by poppastring** [comment]
@ScottIsAFool Gonna be honest, I am struggling with a "good" solution for the JumpList problem. Any help would be appreciated.

**Comment by ScottIsAFool** [comment]
What have you got so far? Put it in a separate branch so we don't have to worry about messing this one up :)

**Comment by poppastring** [comment]
Please let me know what you think of the latest change in relation to recent posts, it is much smaller than my original plan (and hopefully works) @ScottIsAFool  @kathweaver

**Comment by kathweaver** [comment]
Well, it gives an error message now that doesn't exactly made sense.

I think either the "jumplist"  should show after the posts are moved, or they should be listed from the new location.

**Comment by poppastring** [comment]
Not seeing an error message, could you attach a print screen? @kathweaver

**Comment by kathweaver** [comment]
I am seeing an error message -- it says that the post doesn't exist because I just moved it.  However, the post does exist.

**Comment by ScottIsAFool** [comment]
Ok, so here's my issue:

1. Launch Live Writer
2. Click File > hover over local drafts, confirm there are some
3. Go to options, change the default folder.
4. Click to confirm moving existing posts
5. Click File > Hover over local drafts, no local drafts are now shown.

**Comment by kathweaver** [comment]
That's correct @ScottIsAFool

**Comment by poppastring** [comment]
@ScottIsAFool Ok, I can replicate that issue, I will work on that one tonight. I still am unable to get "file does not exist error".

**Comment by ScottIsAFool** [comment]
@poppastring I didn't get that either.

**Comment by poppastring** [comment]
@ScottIsAFool Check out the latest update it was fairly small. Thx

**Comment by SteveP-UK** [comment]
@ScottIsAFool <https://github.com/ScottIsAFool> I don't see this option
either. Iive been using 0.5.1.4 and 0.5.1.5

Regards

Steve
www.StevePalmer.com
blog.StevePalmer.com
@TweetsBySteveP
Mobile: 07860 639 661
:::::::::::::::::::::::::::::::::::::::::::::::::::

**Comment by kathweaver** [comment]
The latest update you made fixed the issue I was seeing.  However, now if I try to move my posts to a folder that has posts in it, I get an error message.
[ErrorReport.txt](https://github.com/OpenLiveWriter/OpenLiveWriter/files/125971/ErrorReport.txt)

**Comment by kathweaver** [comment]
@SteveP-UK You need to click on "Show all checks", "Details", "Artifacts', and then download and run the setup.  It shouldn't be in the nightly build until merged.

So here's the process: pull request is created -- we test the pull -- the pull gets merged then it shows in the nightly builds, eventually we'll get a release and it will show in the main website.

**Comment by SteveP-UK** [comment]
Hi @kathweaver Thanks for the explanation of the process - I'm learning a lot from you guys over and above OLW! :) I'm not in the industry so much of the jargon you use is completely alien to me. I'm not sure where I'm supposed to be looking for show all checks.. details... artifacts - can you point me in the right direction. Thanks for your patience. 

Steve

**Comment by kathweaver** [comment]
Right below this post you'll see what I was talking about.

**Comment by SteveP-UK** [comment]
Hi @kathweaver Found it! Thanks so much! :) S

**Comment by kathweaver** [comment]
Take a look at this pull for the documentation:  https://github.com/OpenLiveWriter/OpenLiveWriter.Github.io/pull/52
It details what I have said here.

**Comment by poppastring** [comment]
My latest changes account for handling files with the same name in both folders when you actually move folders. @kathweaver @ScottIsAFool

**Comment by kathweaver** [comment]
Good news.  Your latest changes fixed the problems I was having.  About to go deeper in the program and do a full test.

**Comment by kathweaver** [comment]
Looks good to me!

**Pull request merged**